Add dog kennel

Charyline opened the pull request:
A dog kennel can be made using a big stack of boards on pit stakes, then hammer and nails, then straw, then a white pillow.
It can be dismantled by removing the pillow by hand, using a pitchfork on the roof and using an adze on the rest to get the boards back.
A dog that doesn't have a short shaft in its mouth will at some point become tired. It has several uses before it gets old, then dies. While it's tired, it looks for a dog kennel, then rests inside. After 120 seconds (like cats on pillows), it stands in front of it and looks for a free tile to leave it.

![image](https://github.com/user-attachments/assets/38d94588-ee3d-4fbb-86b3-c59e8b8a3d3b)

Charyline committed:
Add dog kennel and tired dogs as intermediate state before old dogs. Tired dogs go to kennel to rest.

Charyline committed:
Remove added comments on dog names

risvh commented:
Thanks, below are my comments. Pardon my lack of formatting.

19316 - creation sound. Layer order is a bit weird, stake shadow is above board. Cannot remove boards from this.
19314 - both sprites have the black dot in upper left corner. Creation sound should be init only. Destruction of this gives 880, which should really be a justDismantled version of big stack of boards, see 14307.
19315 - subjective but creation sound should be that of 12042. Roof sprite also has black dot. Creation sound should be init only.
19339 - when last use of this object decays into 1749, it is a Chase move instead of Find (like other uses of 19339) or None (like cats)
all tired dogs are lacking animation
all dogs resting and standing in kennel use the dog name. Maybe something like "[Dog name] in Kennel"?
19350 - shouln't have creation sound
all dogs standing in kennel shouldn't have creation sound, or use the creation sound of "[Dog name]#startFetching" (e.g. 12968)
These object files 1657, 1659, 1660, 1661, 1662, 1663, 1664, 1665, 1759, 1760, 1762, 1763, 1764, 1765, 1766, 1767, 1768, 1769, 1770, 1772, 1773, 1774, 1775, 1776, 1777, 1778 have been changed and some have their name changed back in commit be5c3a130bc20462196f09dcdb6c3b119ae1abde. However, these object files are ancient, so the editor added some additional unnecessary lines to them as it touches them. Consider discarding the changes to these files.

Charyline committed:
Fix top left black dot in kennel sprites

Charyline committed:
Add justDismantled big stack of boards and use it in kennel dismantlment. Fix sounds on kennel objects and dogs standing in front of kennels. Fix transition of tired dog LT to old dog to find behavior. Fix name of resting dogs to dogName in kennel.

Fix dog in kennel sound and change kennel creation sound to initial only

Charyline committed:
Add animations for tired dogs and dogs resting. Change resting german
shepherd object to have paw layer under tail.

Charyline commented:
Creation sounds should be fixed, I added the justDismantled boards and the transition to remove them from the pit stakes, re-added the sprites using a white background and it successfully removed the black dots. The creation sounds should be fixed.

I also set the kennel creation sound to initialOnly because of the decay transition from justLeft.
I added animations on the dogs so that they rest their head lower then tilt it up again, and the tail comes closer to the body. It's my first time animating anything, I hope the result is ok.
I added the yapping sound not only to dogs standing in front of the kennel (when leaving it) but also when going into a kennel, because the straw sound didn't really match.

Let me know if you think I should change it back to straw sound or something else, and if I should add a transition for the pitchfork with straw to put the straw back on the kennel (I saw you try that yesterday).

The only thing left is to restore the files that were changed. I don't really know how to do it with git, so I'll have to find that out or do it manually.

Charyline committed:
Reverted dog files that were changed unnecessarily"

risvh committed:
Fix sound of big stack of boards from dismantling kennel

risvh committed:
More name changes

risvh commented:
- You have 19485 decays into 880 but 880 doesn't have "initial only" so it plays its creation sound after decaying from 19485. That being said, if you wanna make 880 "initial only", you better cross check all the transition that makes 880, see if the change messes up other transitions. (This is just fyi, I've fixed this already)
- The animation looks alright. For future reference, there are downsides to having animations, it takes up resources, and it blurs the crisp lines of the sprites (same goes for rotation of sprites that isn't a multiple of 45 deg), which is especially apparent when the animation is slow. And the upside is also a bit diminished by the fact that most dog breeds in kennel have their wiggling tails hidden behind the kennel wall. 
- Don't mind about pitchfork. Allow pitchfork to act as straw is as bad as allowing both small cut stones and basket of small cut stones on all their transitions. It's unnecessary and it adds to maintenance difficulty. Sidenote, pitchfork is one of the things that may get overhauled later on.

Great job overall, thanks! Will merge later.

Charyline commented:
We could probably remove the animations from the tail on most dogs because it's not showing. I think the german shepherd animation looks nice, but sadly, nearly all the dogs are too small to benefit from it.

risvh committed:
Creation sound of big stack of boards in pit stakes
